fix(appimage): launch updater with bundled linker

getchoo opened the pull request:
This ensures that our updater~~external processes (including our updater and
Minecraft itself)~~ maintains the same compatibility guarantees as the main
binary

## TODO

- ~~Refactor all uses of QProcess with the new `FS::createProcess()`~~
- ~~Refactor manual AppImage/self-contained detection to use `FS::runningInAppImage()` and `FS::runningSelfContained()`~~

getchoo committed:
fix(appimage): launch external processes with bundled linker

This ensures that external processes (including our updater and
Minecraft itself) maintain the same compatibility guarantees as the main
binary

Signed-off-by: Seth Flynn <getchoo@tuta.io>

TheKodeToad commented on launcher/FileSystem.cpp:
```diff
@@ -895,6 +919,17 @@ QString getApplicationsDir()
+bool runningSelfContained(QCoreApplication& app) {
+    const auto appFilePath = app.applicationFilePath();
+    return QFileInfo(appFilePath).fileName().startsWith("ld-linux");
+}
```
These are static
```suggestion
bool runningInAppImage() {
    return QCoreApplication::applicationFilePath().startsWith("/tmp/.mount_");
}

bool runningSelfContained() {
    const auto appFilePath = QCoreApplication::applicationFilePath();
    return QFileInfo(appFilePath).fileName().startsWith("ld-linux");
}
```

TheKodeToad commented on launcher/FileSystem.h:
```diff
@@ -358,6 +369,14 @@ QString getDesktopDir();
+
+// Check if we're running inside a self-contained Linux bundle
+bool runningSelfContained(QCoreApplication& app);
+#endif
```
DesktopServices.h is where we have other similar things, it's a little weird to suddenly start putting them in FileSystem

getchoo replied:
:true:

I just put it here because I read desktop

DioEgizio commented on launcher/FileSystem.cpp:
```diff
@@ -772,6 +773,32 @@ QString ResolveExecutable(QString path)
+        qDebug() << "Wrapping" << program << "with self-contained linker";
+
+        QStringList wrappedArguments;
+        wrappedArguments << "--cache-inhibit" << program;
```
do you mean --inhibit-cache ?

getchoo committed:
fix(updater): re-use `binPath` to determine launcher binary path

Previously this would use the un-adjusted value from Qt, allowing it to
fail in edge cases like self-contained AppImages

Signed-off-by: Seth Flynn <getchoo@tuta.io>

DioEgizio reviewed (approved):
<img width="537" height="323" alt="image" src="https://github.com/user-attachments/assets/4e74a6d8-56ce-40e2-9902-1a7a06a84757" />
🎉
